recipes/xgboost/meta.yaml

```diff
@@ -0,0 +1,64 @@
+{% set name = "xgboost" %}
+{% set version = "0.6a2" %}
+{% set sha256 = "ebc4e2bf8c8266212e342ff8ec4f6ae469e8c41a05d099b6778de8424ce32563" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+  patches:
+    - Makefile.patch  # [osx]
```

Comment:
So I don't think that is a good fix. The main reason I say this is `clang++` will try to link to `libc++` whereas `g++` will try to link to `stdlibc++`. On Mac, the latter hasn't been updated for a long time. It comes from `gcc` 4.2.1 (last `gcc` version under GPL2). Also after 10.7 they added `clang` and started moving people to `libc++`. By 10.9 `clang` is the default toolchain on Mac, but they keep `gcc` for backwards compatibility mainly. Also in conda-forge we make use of `clang` extensively on Mac given it is the default toolchain. So there can be compatibility issues one needs to think about with C++ code.